scripts/run_max_ent.py

```diff
@@ -0,0 +1,282 @@
+"""Script running the MaxEnt IRL
+algorithm on a trees data set.
+Usage:
+    python scripts/run_max_ent.py data=$PWD/path_to_trees.csv n_action=5
+Note:
+    The data path must be absolute not relative.
+"""
+import dataclasses
+import logging
+from typing import Any, Dict, Optional
+
+import anytree
+import pandas as pd
+import numpy as np
+
+import phirl.api as ph
+import phirl.hydra_utils as hy
+
+
+# data = pd.read_csv("tree_df.csv", delimiter=",")
+# state_combinations = [list(i) for i in itertools.product([0, 1], repeat=5)]
+# n_states = 32
+# n_action = 5
+
+
+def get_all_trajectory_states_features(trees):
+    """
+    Args:
+        trees: a dataframe with tree information with column names: [Patient_ID, Tree_ID, Node_ID, Mutation_ID, Parent_ID]
+
+    Returns:
+    all_trajectory: a list of trajectories that includes nodes and mutations from root to leaf
+    all_states: a list of all terminal states for each trajectory
+    all_features: a list that records whether a state (including transit state) exists in a trajectory
+    all_transition: a 2D list that records the order of states in each trajectory
+    (e.g. all_transition[0][0] --> the initial state of the first trajectory)
+    """
+    all_trajectory = []
+    all_states = []
+    all_features = []
+    all_transitions = []
+
+    for i in range(1, len(trees) + 1):
```

Comment:
I think it actually should be replaced with iteration over `trees.items()`, as it currently assumes that the trees are numbered from 1 onwards. (Which doesn't need to be the case in general).